src/network/netdev/wireguard.c

```diff
@@ -880,6 +924,45 @@ static int wireguard_read_private_key_file(Wireguard *w) {
         return 0;
 }
 
+static int wireguard_read_preshared_key_file(WireguardPeer *peer) {
+        _cleanup_free_ char *contents = NULL;
+        _cleanup_free_ void *key = NULL;
+        size_t size, key_len;
+        NetDev *netdev;
+        int r;
+
+        assert(peer);
+        netdev = NETDEV(peer->wireguard);
+        assert(netdev);
+
+        if (!peer->preshared_key_file)
+                return 0;
+
+        r = read_full_file(peer->preshared_key_file, &contents, &size);
+        if (r < 0)
+                return log_netdev_error_errno(netdev, r,
+                                              "Failed to read preshared key from '%s'. "
+                                              "Ignoring [WireGuardPeer] section in %s:%u.",
+                                              peer->preshared_key_file,
+                                              peer->section->filename, peer->section->line);
+
+        r = unbase64mem(contents, size, &key, &key_len);
+        if (r < 0)
+                return log_netdev_error_errno(netdev, r,
+                                              "Failed to decode preshared key. "
+                                              "Ignoring [WireGuardPeer] section in %s:%u.",
+                                              peer->section->filename, peer->section->line);
+
+        if (key_len != WG_KEY_LEN)
+                return log_netdev_error_errno(netdev, SYNTHETIC_ERRNO(EINVAL),
+                                              "Wireguard preshared key has invalid length (%zu bytes)."
+                                              "Ignoring [WireGuardPeer] section in %s:%u.",
+                                              key_len, peer->section->filename, peer->section->line);
```

Comment:
we could probably add a helper that combines read_full_file() + unbase64men() + the key_len check into one, after all we have this at multiple places now?